File: Homeworks/Homework1.Tests/Program.Tests.cs
using Microsoft.VisualStudio.TestPlatform.TestHost;
using Xunit;

namespace Homework1.Tests
{
    public class ProgramTests
    {
        [Fact]
        public void Program_CantParseArguments_1Returned()
        {
            var res = Program.Main(new[] {"10", "+", "z"});
            Assert.Equal(1,res);
        }
        [Fact]
        public void Program_CantParseOperation_2Returned()
        {
            var res = Program.Main(new[] {"10", "?", "12"});
            Assert.Equal(2,res);
        }
        [Fact]
        public void Program_CorrectArgumentsAndOperation_0Returned()
        {
            var res = Program.Main(new[] {"10", "*", "12"});
            Assert.Equal(0,res);
        }
    }
}

File: Homeworks/Homework5/Homework5.Tests/Program.Tests.fs
module Homework5.Tests.Program_Test

open Homework5
open Xunit

[<Fact>]
let ``program_CorrectValues_ZeroReturned``() =
    let result = Program.main [|"12";"+";"3"|]
    Assert.Equal(0,result)
    
let args =
    [|
        [|"x";"-";"2"|]
        [|"10";"+";"z"|]
        [|"x";"+";"z"|]
        [|"11";"x";"123"|]
        [|"12";"/";"0"|]
    |]
    
[<Fact>]
let ``program_InvalidArguments_Error1Returned``() =
    args |> Array.iter(fun arg-> Assert.Equal(Program.main arg,1))
